subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -498,43 +627,21 @@ int img_mgmt_set_next_boot_slot(int slot, bool confirm)
 		/* Allow confirming slot == active_slot */
 	}
 
-	if (flash_area_open(area_id, &fa) != 0) {
-		return IMG_MGMT_ERR_FLASH_OPEN_FAILED;
-	}
-
-	rc = boot_set_next(fa, slot == active_slot, confirm);
-	if (rc != 0) {
-		/* Failed to set next slot for boot as desired */
-		LOG_ERR("Faled boot_set_next with code %d, for slot %d,"
-			" with active slot %d and confirm %d",
-			 rc, slot, active_slot, confirm);
-
-		/* Translate from boot util error code to IMG mgmt group error code */
-		if (rc == BOOT_EFLASH) {
-			rc = IMG_MGMT_ERR_FLASH_WRITE_FAILED;
-		} else if (rc == BOOT_EBADVECT) {
-			rc = IMG_MGMT_ERR_INVALID_IMAGE_VECTOR_TABLE;
-		} else if (rc == BOOT_EBADIMAGE) {
-			rc = IMG_MGMT_ERR_INVALID_IMAGE_HEADER_MAGIC;
-		} else {
-			rc = IMG_MGMT_ERR_UNKNOWN;
-		}
-	}
-	flash_area_close(fa);
-
-#if defined(CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS)
-	if (slot == active_slot && confirm) {
-		int32_t err_rc;
-		uint16_t err_group;
+	return img_mgmt_set_next_boot_slot_common(slot, active_slot, confirm);
+}
+#else
+int img_mgmt_set_next_boot_slot(int slot, bool confirm)
+{
+	int active_image = img_mgmt_active_image();
```

Comment:
Shouldn't we use `img_mgmt_slot_to_image` here? We could return error if it's not the same as `img_mgmt_active_image` if needed.
With current implementation it should not lead to big problems since XIP with revert is not supported with multiple images, but it may be counterintuitive.

Reply:
With the current allowed configuration where direct xip support only one image that would check that is redundant as image other than 0 should never get here.

Reply:
Then I would suggest to consider asserting this (to easily catch potential regressions)

Reply:
No assertion here then?